### ISSUE TITLE
No Support for Compile Time DI with Play 2.4

### ISSUE THREAD

**Issue by loicdescotte** [playframework/play-mailer#69]
Hi,
Play 2.4 supports both runtime and compile time  dependency injection.   
As CommonsMailer uses `@Inject` it seems impossible to use it without the Guice container.
It would be nice to provide a trait that provide a "raw" mailer client and that could be mixed in Play's `BuiltInComponentsFromContext` to support compile time DI, like [this WS Component](https://www.playframework.com/documentation/2.4.x/api/scala/index.html#play.api.libs.ws.ning.NingWSComponents) for example.

Thanks

Loïc

**Comment by Mogztter** [comment]
Hi @loicdescotte,

With this [pull request](https://github.com/playframework/play-mailer/pull/63), you will be able to manually create a mailer client:
```scala
val email = Email("Simple email", "Mister FROM <from@email.com>")
new SMTPMailer(SMTPConfiguration("typesafe.org", 1234)).send(email)
```

AFAIK `NingWS` also use `@Inject`, for example in the `NingWSAPI` [constructor](https://github.com/playframework/playframework/blob/master/framework/src/play-ws/src/main/scala/play/api/libs/ws/ning/NingWS.scala#L380) (i.e. the dependency on Guice is provided by Play!). So basically we just need to create a trait ?

```scala
trait MailerComponents {

  def configuration: Configuration

  lazy val mailerClient = new SMTPMailer(new SMTPConfigurationProvider(configuration).get())
}
```

Having said that, we still need to sort out how to let user provide their own `SMTPConfiguration` and how to provide multiple mailer clients (for other email service providers as requested in https://github.com/playframework/play-mailer/issues/10)

**Comment by loicdescotte** [comment]
@Mogztter great, I hope it will be merged :)
Indeed I guess we just need to create a trait like this one.

**Comment by Mogztter** [comment]
Resolved by #70
